### PR TITLE
Warn when usage of a deprecated field is found

### LINKED ISSUE #73: Trigger a compilation warning when using a deprecated field
It's important for us to know when we are using field that's been deprecated so we can keep up to date with the backend.

### PR DESCRIPTION
**Pull request by romanschejbal**
Fixes #73 

It's still a work in progress as I'm not sure how to write the test for this since I'm unfamiliar with accessing the context from within the test scope. Could you maybe help/guide me here a bit, please? @baransu

### CODE REVIEW

**Comment by baransu** [comment]
Not sure if it's possible to access the context in tests at the moment. I think it's ok to just have that in tests as it would be a lot of work to make assertions on warning/errors.

**Comment by baransu** [comment]
I'll try to review soon!

File: src/bucklescript/graphql_ppx.re [hunk 1]
```diff
@@ -96,7 +96,14 @@ let rewrite_query = (~schema=?, ~loc, ~delim, ~query, ()) => {
         /*  the only call site of schema, make it lazy! */
         schema: Lazy.force(Read_schema.get_schema(schema)),
       };
-      switch (Validations.run_validators(config, document)) {
+      let (errors, warnings) = Validations.run_validators(config, document);
+      warnings
+      |> List.iter(((loc, msg)) => {
```
Comment (baransu) [inline src/bucklescript/graphql_ppx.re]: This wouldn't work. `List.iter` expexts unit but you're retuning expression here.

Another thing is - to show a warning to a user we have to add it to `Pmod_structure`.
There is a difference between errors and a valid query. I think in warning case you have to construct
`Ast_mapper.attribute_of_warning` (http://caml.inria.fr/pub/docs/manual-ocaml/compilerlibref/Ast_mapper.html) expression and add it to both valid and invalid module to show warnings all the time.

File: src/native/graphql_ppx.re [hunk 2]
```diff
@@ -111,7 +111,14 @@ let rewrite_query = (~schema=?, ~loc, ~delim, ~query, ()) => {
         /*  the only call site of schema, make it lazy! */
         schema: Lazy.force(Read_schema.get_schema(schema)),
       };
-      switch (Validations.run_validators(config, document)) {
+      let (errors, warnings) = Validations.run_validators(config, document);
+      warnings
+      |> List.iter(((loc, msg)) => {
```
Comment (baransu) [inline src/native/graphql_ppx.re]: Same applies here.

**Comment by jfrolich** [comment]
Replaced by: #117

**Pull request closed without merging**